core/src/main/java/org/apache/iceberg/rest/RESTClient.java

```diff
@@ -30,6 +30,6 @@
   <T> T delete(String path, Class<T> responseType, Consumer<ErrorResponse> errorHandler);
   <T> T post(String path, Object body, Class<T> responseType, Consumer<ErrorResponse> errorHandler);
   <T> T get(String path, Class<T> responseType, Consumer<ErrorResponse> errorHandler);
-  <T> T head(String path, Consumer<ErrorResponse> errorHandler);
+  void  head(String path, Consumer<ErrorResponse> errorHandler);
```

Comment:
We determined that `HEAD` requests don't return a response. I'm still open to allowing them to though (as many well tested systems, including S3, do use a response b ody for HEAD).

Reply:
I think it's fine either way.